src/lib/tag-parsers/generate-page-auto-data.js

```diff
@@ -0,0 +1,43 @@
+import parseOgTags from './og-parser';
+import parseJsonLd from './jsonld-parser';
+import { _get } from "../utils";
+
+const generateOgData = () => {
+    const ogTags = parseOgTags();
+    const hasOgTags = Array.isArray(ogTags) && ogTags.length > 0
+    return hasOgTags ? {
+        type: "open_graph",
+        version: 1,
+        data: ogTags
+    } : null;
+}
+
+const generateJSONldData = () => {
+    const ldTags = parseJsonLd();
```

Comment:
Please keep in mind that we need to send/parse only limited number of json ld structures. https://schema.org/Product and https://schema.org/BreadcrumbList

Reply:
Has to be limited since the kafka message has a max payload size.